posthog/async_migrations/setup.py

```diff
@@ -119,3 +123,39 @@ def get_async_migration_dependency(migration_name: str) -> Optional[str]:
         return None
 
     return ASYNC_MIGRATION_TO_DEPENDENCY[migration_name]
+
+
+def verify_required_migrations_have_been_completed(ignore_posthog_version: bool, applied_migrations: set):
+    unapplied_migrations = set(ALL_ASYNC_MIGRATIONS.keys()) - applied_migrations
+    for migration_name, migration in ALL_ASYNC_MIGRATIONS.items():
+        if (
+            (not ignore_posthog_version)
+            and (migration_name in unapplied_migrations)
+            and (POSTHOG_VERSION > Version(migration.posthog_max_version))
+        ):
+            raise ImproperlyConfigured(f"Migration {migration_name} is required for PostHog versions above {VERSION}.")
+
+
+def is_fresh_install(first_migration: str):
+    # To verify we're on a fresh install
+    # (1) check that the first async migration is not started
+    # (2) check that no persons exist (i.e. someone didn't try to upgrade to 1.40 from before async migrations introduction)
+    migration_instance = AsyncMigration.objects.filter(name=first_migration).first()
+    if migration_instance.status != MigrationStatus.NotStarted:
+        return False
+    event_cnt = sync_execute("SELECT count() FROM events")[0][0]
+    person_cnt = sync_execute("SELECT count() FROM person")[0][0]
+    if event_cnt > 0 or person_cnt > 0:
+        return False
+    return True
+
+
+def mark_all_existing_migrations_as_completed():
+    logger.info("Fresh install - marking all existing migrations as completed")
+    # we can tell by the percentage being 0 and time being the same that they were completed at startup
```

Comment:
I don't like the idea of progress showing 0 - will make me feel like something is amiss.

Are there other ways we can do this? Maybe a new status `CompletedAutomatically`, `Skipped`, or `Autocompleted`.

Same finish time already might be enough too

Reply:
I like the idea of the new status